dask/dataframe/io/parquet/arrow.py

```diff
@@ -651,10 +628,33 @@ def initialize_write(
     def _pandas_to_arrow_table(
         cls, df: pd.DataFrame, preserve_index=False, schema=None
     ) -> pa.Table:
-        table = pa.Table.from_pandas(
-            df, nthreads=1, preserve_index=preserve_index, schema=schema
-        )
-        return table
+        try:
+            return pa.Table.from_pandas(
+                df, nthreads=1, preserve_index=preserve_index, schema=schema
+            )
+        except pa.ArrowException as exc:
+            if schema is None:
+                raise
```

Comment:
`ArrowException` is very broad (I think I've also seen it with datetime coercion issues, for instance. Maybe it's also worth sniffing the message here to make sure there isn't another reason for the exception before showing a possibly-wrong `ValueError`?

Reply:
Yes, I agree that any arrow error is relevant: I'm worried about false positives for the schema mismatch message. What if there is a relevant `ArrowException` that is not schema-related while `schema="infer"`?

Reply:
I'm not sure I follow? If we get an error here, it's because we failed to convert the pandas dataframe to a pyarrow table with the specified schema. If so, we raise a message stating this, with the original error message, and then display both the expected and actual partition schemas. I think this should be enough information for a user to infer what's going on, even if we are accidentally over eager in blaming this on a schema mismatch.

Would a check like below resolve your worries here?

```python
if schema == df_schema:
    raise  # not a schema mismatch
```